### PR TITLE
Add pre-commit configuration

### PR DESCRIPTION
**Pull request by knightdave**
related https://github.com/insightsengineering/automation-tasks/issues/28

### CODE REVIEW

File: inst/WORDLIST [hunk 1]
```diff
@@ -1,3 +1,135 @@
...
+grepl
+gsub
+gz
+href
+http
 https
+httr
+ℹ
+indentify
+informatoin
```
Comment (tomszosz) [inline inst/WORDLIST:53]: this is typo

Comment (knightdave) [reply inst/WORDLIST]: fixed

File: README.md [hunk 2]
```diff
@@ -4,40 +4,142 @@ An R package template with built-in Github Actions-based CI/CD workflows.
...
 
-* Update the remainder of the package as you would do typically while developing an R package.
+- Update the remainder of the package as you would do typically while developing an R package.
 
-* Enjoy!
+- Enjoy!
+
+## Pre-commit
+
+This repository contains example [pre-commit] configuration.
```
Comment (dinakar29) [inline README.md:47]:
```suggestion
This repository contains an example [pre-commit] configuration.
```

File: README.md [hunk 3]
```diff
@@ -4,40 +4,142 @@ An R package template with built-in Github Actions-based CI/CD workflows.
...
+
+[pre-commit] is a tool that use [Git hooks] to identify and solve simple issues before submission to code review.
```
Comment (dinakar29) [inline README.md:49]:
```suggestion
[pre-commit] is a tool that uses [Git hooks] to identify and resolve simple issues prior to submission for code review.
```

File: README.md [hunk 4]
```diff
@@ -4,40 +4,142 @@ An R package template with built-in Github Actions-based CI/CD workflows.
...
+[Git hooks] run on every commit to automatically point out and solve issue like missing semicolons, trailing whitespace,
```
Comment (dinakar29) [inline README.md:50]:
```suggestion
[Git hooks] run on every commit to automatically point out and solve issues such as missing semicolons, trailing whitespaces,
```

File: README.md [hunk 5]
```diff
@@ -4,40 +4,142 @@ An R package template with built-in Github Actions-based CI/CD workflows.
...
+code formatting and spell checks.
+
+### Setting up pre-commit for R project
+
+1. Install pre-commit framework. Use official [installation guide][pre-commit installation].
```
Comment (dinakar29) [inline README.md:55]:
```suggestion
1. Install the `pre-commit` framework. Use the official [installation guide][pre-commit installation].
```

File: README.md [hunk 6]
```diff
@@ -4,40 +4,142 @@ An R package template with built-in Github Actions-based CI/CD workflows.
...
+```
+
+3. Run script to generate example pre-commit configuration:
+
+```sh
+R -e 'precommit::use_precommit()'
+```
+
+Above command will generate example `.pre-commit-config.yaml`. If this file already exists in repo
+you can skip this step.
```
Comment (dinakar29) [inline README.md:69]:
```suggestion
```

File: README.md [hunk 7]
```diff
@@ -4,40 +4,142 @@ An R package template with built-in Github Actions-based CI/CD workflows.
...
+2. Install R package `precommit`
+
+```sh
+R -e 'install.packages("precommit")'
+```
+
+3. Run script to generate example pre-commit configuration:
+
+```sh
+R -e 'precommit::use_precommit()'
```
Comment (dinakar29) [inline README.md:65]:
```suggestion
[ ! -f ".pre-commit-config.yaml" ] && R -e 'precommit::use_precommit()'
```

File: README.md [hunk 8]
```diff
@@ -4,40 +4,142 @@ An R package template with built-in Github Actions-based CI/CD workflows.
...
+### Setting up pre-commit for R project
+
+1. Install pre-commit framework. Use official [installation guide][pre-commit installation].
+2. Install R package `precommit`
+
+```sh
+R -e 'install.packages("precommit")'
+```
+
+3. Run script to generate example pre-commit configuration:
```
Comment (dinakar29) [inline README.md:62]:
```suggestion
3. Run the `use_precommit()` function to generate an example pre-commit configuration called `.pre-commit-config.yaml`:
```

File: README.md [hunk 9]
```diff
@@ -4,40 +4,142 @@ An R package template with built-in Github Actions-based CI/CD workflows.
...
+Check Yaml...............................................................Passed
+Don't commit to branch...................................................Passed
+Mixed line ending........................................................Passed
+Don't commit common R artifacts......................(no files to check)Skipped
+
+```
+
+[pre-commit]: https://pre-commit.com
+[pre-commit installation]: https://pre-commit.com/#installation
+[git hooks]: https://git-scm.com/book/en/v2/Customizing-Git-Git-Hooks
```
Comment (dinakar29) [inline README.md:145]: Move links to top of the README.

File: .pre-commit-config.yaml [hunk 10]
```diff
@@ -0,0 +1,67 @@
...
+        exclude: >
+          (?x)^(
+          data/.*|
+          (.*/|)\.Rprofile|
+          (.*/|)\.Renviron|
+          (.*/|)\.gitignore|
+          (.*/|)NAMESPACE|
+          (.*/|)WORDLIST|
+          (.*/|)\.travis.yml|
+          (.*/|)appveyor.yml|
```
Comment (dinakar29) [inline .pre-commit-config.yaml:26]: We don't really use these, so we can remove them. Also, let's exclude the `.github` directory.

File: inst/WORDLIST [hunk 11]
```diff
@@ -1,3 +1,130 @@
...
+ubuntu
+usr
+VignetteBuilder
+wget
+www
+xvzf
+yaml
+YAML
+yml
+zricethezav
```
Comment (dinakar29) [inline inst/WORDLIST:130]: A lot of these are from `,github`, so they won't be needed. These are false positives.

**Review by dinakar29 — APPROVED** [review]
LGTM

**Pull request merged**